File: backend/reservation-service/src/main/java/com/hotel/reservation_service/service/PaymentService.java
package com.hotel.reservation_service.service;

import com.hotel.reservation_service.entity.Payment;

import java.util.List;
import java.util.UUID;

public interface PaymentService {
    Payment getPaymentsByReservationId(UUID reservationId);
    Payment createPayment(Payment payment, UUID reservationId);
}

File: frontend/src/app/services/reservation/payment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Payment } from '../../model/payment.model';
import { AppConstants } from "../../config/app.constants";
import { AuthService } from '../auth/auth.service';

@Injectable({
  providedIn: 'root'
})
export class PaymentService {
  private apiUrl = `${AppConstants.BASE_API_V1_URL}/payments`;

  constructor(private http: HttpClient, private authService: AuthService) {}

  createPayment(paymentData: Payment, reservationId: string): Observable<Payment> {
    const headers = new HttpHeaders({
      'Logged-User': this.authService.getUserInformation()[1].value
    });

    return this.http.post<Payment>(`${this.apiUrl}/reservation/${reservationId}`, paymentData, { headers });
  }
}
